### PR TITLE
data FEATURE JSON data format support

### PR DESCRIPTION
**Pull request by rkrejci**

### CODE REVIEW

File: src/tree_data.c [hunk 1]
```diff
@@ -404,16 +452,20 @@ lyd_parse_rpc(const struct ly_ctx *ctx, struct ly_in *in, LYD_FORMAT format, str
     format = lyd_parse_get_format(in, format);
     LY_CHECK_ARG_RET(ctx, format, LY_EINVAL);
 
+    /* init */
+    *tree = NULL;
+    if (op) {
+        *op = NULL;
+    }
+
```
Comment (michalvasko) [inline src/tree_data.c:460]: Allow to set either.

File: src/tree_data.c [hunk 2]
```diff
@@ -460,16 +518,20 @@ lyd_parse_notif(const struct ly_ctx *ctx, struct ly_in *in, LYD_FORMAT format, s
     format = lyd_parse_get_format(in, format);
     LY_CHECK_ARG_RET(ctx, format, LY_EINVAL);
 
+    /* init */
+    *tree = NULL;
+    if (ntf) {
+        *ntf = NULL;
+    }
+
```
Comment (michalvasko) [inline src/tree_data.c:526]: Allow to set either.

Comment (rkrejci) [reply src/tree_data.c:526]: ok

File: src/tree_data.c (hunk 1 above)
Comment (rkrejci) [reply src/tree_data.c:460]: ok

**Pull request merged**